docs/DeveloperGuide.md

```diff
@@ -1,22 +1,222 @@
+while a mealEntry is a concrete calorie consumption the user wants to track. The latter makes a timestamp indispensible. 
+
+#### HealthGoal
+The HealthGoal class manages a user's health goal and calculates target calorie intake based on user data such as height, weight, age, and gender. 
+It offers three main health goals: weight loss, steady state, and bulking, each with a corresponding calorie modification factor. 
+The class supports setting, storing, and retrieving the health goal. 
+Additionally, it provides a method to compute target calories using the Harris-Benedict Equation, modified by the current health goal. 
+This allows the class to adapt the calorie calculations according to the user's health objectives, making it versatile for various fitness plans.
+
+#### User
+The user class encapsulates all necessary information for computing an ideal daily calorie consumption. 
+This includes:
+- Height (in cm): Double 
+- Weight (in cm): Double
+- Age: Integer
+- Gender: Boolean
+- Health goal: HealthGoal
+- Ideal calorie intake: Computed based on the information above 
+- Date: LocalDateTime specifying the date of the above information
+More details follow in the Features section of this guide. 
+
+#### HistoryTracker
+The HistoryTracker class is responsible for managing the persistence of data in the HealthMate application. It handles the saving and loading of user data, meal entries, and meal options to and from files. This class plays a crucial role in maintaining the application's state across different sessions.
+
+Key features of the HistoryTracker class include:
+
+1. Data Persistence:
+   - Saves user data, meal entries, and meal options to separate files.
+   - Loads existing data from these files when a new usage session is initiated.
+
+2. File Management:
+   - Creates and manages the necessary files for storing application data.
+   - Handles file I/O operations, ensuring data integrity during read and write processes.
+
+3. Data Formatting:
+   - Converts complex objects (like User, MealEntry, and Meal) into a format suitable for file storage.
+   - Deserializes stored data back into usable objects when loading the application state.
+
+HistoryTracker allows for the persistance of user inputted data between sessions by storing it in a local csv file.
+
+## Features
+
+This section will document the contributions made by each team member regarding the implementation or planned feature enhancements, detailing the design and thought processes behind them.
+
+---
+### Creating a User Profile
+To create or load a user profile the `User` class provides the method `checkForUserData` which loads 
+saved user information if available from an existing file usingn the `HistoryTracker` or prompts the user
+to input new information for creating a new profile as shown in the sequence diagram below.
+![User SD](images/userSequenceDiagram.jpg)
+Reference diagrams used
+![User SD](images/askForUserDataSD.png)
```

Comment:
in the sequence, "printString(...)" is used multiple times. would it be more readable to differentiate each print string with specific text or parameters? (eg, printstring(enter calorie:)). this could help viewers understand what actual prompt is displayed at each step.